ReactSkia/components/RSkComponentScrollView.cpp

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include "include/core/SkPaint.h"
+#include "include/core/SkPictureRecorder.h"
+
+#include "compositor/layers/PictureLayer.h"
+
+#include "ReactSkia/components/RSkComponentScrollView.h"
+#include "ReactSkia/utils/RnsUtils.h"
+#include "ReactSkia/views/common/RSkDrawUtils.h"
+
+namespace facebook {
+namespace react {
+
+#define SCROLL_NEXT 1
+#define SCROLL_PREV 0
+
+using namespace RSkDrawUtils;
+
+RSkComponentScrollView::RSkComponentScrollView(const ShadowView &shadowView , RSkSurfaceWindow * surface)
+    : RSkComponent(shadowView),
+      containerWidth_(shadowView.layoutMetrics.frame.size.width),
+      containerHeight_(shadowView.layoutMetrics.frame.size.height),
+      surface_(surface) {}
+
+
+void RSkComponentScrollView::updateComponentProps(
+    const ShadowView &newShadowView,
+    bool forceUpadate) {
+
+  auto const &scrollViewProps = *std::static_pointer_cast<ScrollViewProps const>(newShadowView.props);
+
+  isScrollEnabled_ = scrollViewProps.scrollEnabled;
+
+  if( forceUpadate || (contentInset_ != scrollViewProps.contentInset)) {
+       contentInset_ = scrollViewProps.contentInset;
+       forceBitmapReset_ = true;
+  }
+
+}
+
+/* ScrollView childrens are maintained locally by component */
+/* ScrollView childrens are not added to render layer tree */
+void RSkComponentScrollView::mountChildComponent(
+    std::shared_ptr<RSkComponent> newChildComponent,
+    const int index) {
+
+    RNS_LOG_INFO("ScrollView mountChildComponent tag["<< newChildComponent->getComponentData().tag << "] index["<<index << "]");
+
+    childList_.insert(childList_.begin() + index, newChildComponent);
+    forceBitmapReset_ = true;
+
+    Rect childFrame = newChildComponent->getComponentData().layoutMetrics.frame;
+    if((childFrame.origin.y + childFrame.size.height) > containerHeight_)
+         containerHeight_ = childFrame.origin.y + childFrame.size.height + contentInset_.bottom;
+
+    if((childFrame.origin.x + childFrame.size.width) > containerWidth_)
+         containerWidth_ = childFrame.origin.x + childFrame.size.width + contentInset_.right;
+
+    /* TODO Add contentInset bottom and right values */
+    RNS_LOG_INFO("ScrollView container height["<< containerHeight_ << "] width["<< containerWidth_ << "]");
+}
+
+void RSkComponentScrollView::unmountChildComponent(
+    std::shared_ptr<RSkComponent> oldChildComponent,
+    const int index) {
+
+    RNS_LOG_INFO("ScrollView unmountChildComponent tag["<< oldChildComponent->getComponentData().tag << "] index["<<index << "]");
+
+    childList_.erase(childList_.begin() + index);
+    forceBitmapReset_ = true;
+
+    Rect childFrame = oldChildComponent->getComponentData().layoutMetrics.frame;
+    if(!isHorizontalScroll()) {
+       if(childFrame.origin.y < containerHeight_)
```

Comment:
If the scroll content are placed with spacing using margin or padding. then resetting the height to the teh removed component origin will leave some space between the last element & the end of container. Is that the expected behavior or to be considered

Reply:
Can a element in the middle of the scroll view container be removed. If possible, will the container dimension be reset to the middle, ignoring the further element in the container...

Reply:
Good point 👍 Yes this is possible. This logic will have issue in this case. 
Will update it to handle this as well.